app/uk/gov/hmrc/gform/sharedmodel/VariadicFormData.scala

```diff
@@ -118,6 +119,13 @@ case class VariadicFormData[S <: SourceOrigin](data: Map[ModelComponentId, Varia
       }
     }
 
+  def distinctIndexedComponentIds(modelComponentId: ModelComponentId): List[IndexedComponentId] =
+    forBaseComponentIdLessThen(modelComponentId)
+      .map(_._1.fold(p => p.indexedComponentId)(i => i.indexedComponentId))
```

Comment:
```suggestion
   forBaseComponentId(modelComponentId.indexedComponentId.baseComponentId)
      .map { case (modelComponentId, _) => modelComponentId.fold(p => p.indexedComponentId)(i => i.indexedComponentId) }
```
`forBaseComponentIdLessThen` should not be used here, we are not in ATL. Also let's get rid of `_._1`